lib/sycamore/sycamore/executor.py

```diff
@@ -87,6 +89,23 @@ def execute_iter(self, plan: Node, **kwargs) -> Iterable[Document]:
 
         plan.traverse(visit=lambda n: n.finalize())
 
+    def _prepare(self, plan: Node):
+        from queue import Queue
+
+        pending: Queue[Callable] = Queue()
+
+        def visit(n):
+            f = n.prepare()
+            if f is not None:
+                pending.put(f)
+
+        plan.traverse(visit=visit)
+        while not pending.empty():
+            f = pending.get(block=False)
+            g = f()
+            if g is not None:
+                pending.put(g)
```

Comment:
Per discussion no.  If you clean and then check, the check is pointless (you just cleaned). If you check and then clean, any time you run the same script twice it will fail (because the clean file will be present from the previous run). So multiple passes are needed.